### PR TITLE
Implement possible fragment spreads validation rule

### PR DESCRIPTION
**Pull request by rawls238**
@dittos

### CODE REVIEW

**Comment by jhgg** [comment]
Don't forget to `tox` before you push. It'll run the same checks as travis.

File: graphql/core/validation/rules.py [hunk 1]
```diff
@@ -234,7 +234,52 @@ def unused_fragment_message(fragment_name):
...
+        return frag and type_from_ast(context.get_schema(), frag.type_condition)
+
+    @staticmethod
+    def do_types_overlap(t1, t2):
+        if t1 == t2:
+            return True
+        if isinstance(t1, GraphQLObjectType):
+            if isinstance(t2, GraphQLObjectType):
+                return False
+            return t2.get_possible_types().index(t1) != -1
```
Comment (dittos) [inline graphql/core/validation/rules.py]: `list.index` throws if there's no element found. Change to `t1 in t2.get_possible_types()`?

File: tests/core_validation/test_possible_fragment_spreads.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  }
+
+
+def error_annon(frag_name, parent_type, frag_type, line, column):
+  return {
+    'message': PossibleFragmentSpreads.type_incompatible_anon_spread_message(frag_name, parent_type, frag_type),
+    'locations': [{ 'line': line, 'column': column}]
+  }
+
+def same_object():
```
Comment (dittos) [inline tests/core_validation/test_possible_fragment_spreads.py]: pytest only runs functions with `test_` prefix. http://pytest.org/latest/goodpractises.html#test-discovery

Comment (rawls238) [inline tests/core_validation/test_possible_fragment_spreads.py]: hm that's strange, I swear that they were running locally but apparently they weren't...

**Comment by rawls238** [comment]
@dittos @jhgg I'm not going to have time to circle back to this until Tuesday / Wednesday so feel free to pick it up if you wish, there were a few failing tests that I couldn't get to the bottom of

**Comment by dittos** [comment]
I'm looking in this now

**Pull request merged**